accounting/on_chain.go

```diff
@@ -30,18 +34,85 @@ func OnChainReport(ctx context.Context, cfg *OnChainConfig) (Report, error) {
 // the actual price API.
 func onChainReportWithPrices(cfg *OnChainConfig, getPrice usdPrice) (Report,
 	error) {
+
+	// Create an info struct to hold all the elements we need.
+	info := onChainInformation{
+		priceFunc:      getPrice,
+		openedChannels: make(map[string]channelInfo),
+		sweeps:         make(map[string]bool),
+		closedChannels: make(map[string]closedChannelInfo),
+	}
+
 	onChainTxns, err := cfg.OnChainTransactions()
 	if err != nil {
 		return nil, err
 	}
 
 	// Filter our on chain transactions by start and end time. If we have
 	// no on chain transactions over this period, we can return early.
-	filtered := filterOnChain(cfg.StartTime, cfg.EndTime, onChainTxns)
-	if len(filtered) == 0 {
+	info.txns = filterOnChain(cfg.StartTime, cfg.EndTime, onChainTxns)
+	if len(info.txns) == 0 {
 		return Report{}, nil
 	}
 
+	// Get our pending channels so that we do not miss channel transactions
+	// that may have confirmed on chain, but are still considered pending
+	// by lnd (this is the case for channel opens that require more than one
+	// conf, or closing channels that are awaiting resolution).
+	pending, err := cfg.PendingChannels()
+	if err != nil {
+		return nil, err
+	}
+
+	// We add our pending force close channels to opened and closed channels
+	// because it is possible that our channel was opened and closed in the
+	// our relevant period.
+	for _, c := range pending.PendingForceClose {
+		inf := newChannelInfo(
+			lnwire.NewShortChanIDFromInt(0), c.ChannelPoint,
+			c.PubKeyBytes, c.Capacity,
+		)
+
+		info.openedChannels[c.ChannelPoint.Hash.String()] = inf
+		info.closedChannels[c.CloseTxid.String()] = closedChannelInfo{
+			channelInfo:    inf,
+			closeType:      "force close",
+			closeInitiator: "unknown for pending channels",
+		}
+	}
+
+	// Add our channel open and all possible channel closes to our info set.
+	// We add all potential close txids in case one of them has confirmed.
+	for _, c := range pending.WaitingClose {
+		inf := newChannelInfo(
+			lnwire.NewShortChanIDFromInt(0), c.ChannelPoint,
+			c.PubKeyBytes, c.Capacity,
+		)
+
+		info.openedChannels[c.ChannelPoint.Hash.String()] = inf
+
+		closed := closedChannelInfo{
+			channelInfo:    inf,
+			closeType:      lndclient.CloseTypeCooperative.String(),
+			closeInitiator: lndclient.InitiatorUnrecorded.String(),
+		}
+		info.closedChannels[c.LocalTxid.String()] = closed
+		info.closedChannels[c.RemoteTxid.String()] = closed
+		info.closedChannels[c.RemotePending.String()] = closed
+	}
+
+	// Add our pending open channel to our set of open channels to cover
+	// the case where our transaction has confirmed on chain but has not
+	// yet reached the confirmations lnd requires.
```

Comment:
are you somewhere filtering out non-confirmed channels? The comment makes it sound like they are not included.

Reply:
yes, these opened/closed lists are just build up to match with the output of `listchaintxns`, which we do filter for only confirmed txns. But I see how the comment can be confusing, since that context isn't very clear here, will rephrase!